### DISCUSSION TITLE
Are all current math methods in the System library going to be updated with the new generic math support?

### DISCUSSION THREAD

**Discussion by AnteaterStudios** [dotnet/runtime#74466] [General] [answered]
Are all current math methods in the System library going to be updated with the new generic math support? I see in the documentation of several methods in the language's library the usage of various possibilities of numeric input parameters. Are those going to be updated to use the new INumber interface? I was just curious, because I imagine that would be a lot of work...

**Comment by PathogenDavid** [comment, marked as answer]
Most (all?) of the `System.Math` APIs are now directly available on applicable types. So you're expected to use those in a generic math context rather than `System.Math`.

For example, calling `Sqrt` generically looks like:

```csharp
TNumber MySqrt<TNumber>(TNumber x)
    where TNumber : IRootFunctions<TNumber>
    => TNumber.Sqrt(x);
```

My understanding is that going forward `System.Math`/`MathF` will be considered legacyish and any new math operations will be added to numeric types directly. This goes for non-generic contexts too. (IE: `double.Sqrt` instead of `Math.Sqrt`)

See also: https://github.com/dotnet/runtime/issues/63732

(cc @tannergooding)

**Reply by tannergooding** [reply]
Yep, this is correct.

**Reply by AnteaterStudios** [reply]
Thanks for the answer! I'm already looking forward to the new generic math feature!

**Reply by danmoseley** [reply]
@AnteaterStudios have you been trying it? you don't need to wait for final release. there are up to date builds listed at https://github.com/dotnet/installer/

**Reply by AnteaterStudios** [reply]
@danmoseley thanks, I will give it a try! Nevertheless, I'm still going to wait for the final release of .NET 7 to include it in my personal project.
